app/assets/stylesheets/govuk_publishing_components/components/_layout-super-navigation-header.scss

```diff
@@ -295,11 +317,29 @@ $button-pipe-colour: darken(govuk-colour("mid-grey"), 20%);
           background: none;
         }
       }
-      // stylelint-enable max-nesting-depth
 
       &:after {
-        @include pseudo-underline($left: govuk-spacing(5), $right: govuk-spacing(6));
+        // pseudo-underline will have the same width as the the
+        // text of the link if js has not been initialised
+
+        // $bottom is the 1px border of .gem-c-layout-super-navigation-header
+        // multiplied by -1
+        @include pseudo-underline($left: govuk-spacing(6), $right: 0, $bottom: -1px, $width: calc(100% - 60px));
       }
+
+      .js-module-initialised & {
+        // if js initialised then remove the padding and margin
+        // applied to the links
+
+        padding: 0;
+        margin: 0;
+
+        &:after {
+          @include pseudo-underline($left: govuk-spacing(5), $right: govuk-spacing(6), $width: calc(100% - 55px));
```

Comment:
Would it be tidier to set variables instead of having all these comments? eg (forgive any broken syntax)
```
$after-left-padding: govuk-spacing(5)
$after-right-padding: govuk-spacing(6)
...
padding-left:$after-left-padding
padding-right: $after-right-padding
&:after {
$width: calc(100% - ($after-left-padding + $after-right-padding)
```

This way the calculation is clear without the comments, if you ever have to change the sizing it's just one place to update, and if the design system changes upstream we don't have to fix that 55px.

Reply:
I've re-opened this comment as it seems Max's suggestion above may have been missed. I would concur with Max's suggestion, as it makes the code easier to read and maintain, and additionally avoids [magic numbers](https://css-tricks.com/magic-numbers-in-css/).

Another suggestion is to use the [Sass interpolation](https://sass-lang.com/documentation/interpolation) feature, which avoids the need to create new variables and still avoids the magic numbers problem -- but this can become unwieldy. For example:

```
@include pseudo-underline(
   $left: govuk-spacing(5),
   $right: govuk-spacing(6),
   $width: calc(100% - #{govuk-spacing(5) + govuk-spacing(6)})
);
```